### DISCUSSION TITLE
Add alpha to status2d

### DISCUSSION THREAD

**Discussion by Vinschers** [bakkeby/dwm-flexipatch#280] [Ideas]
Hello!

First of all, thank you for this amazing project.

I was wondering if it would be possible to add support for transparency in text and drawings with status2d. Currently, alpha patch adds transparency to the statusbar window but keeps all text opaque. I was thinking about something like

^c#0000FF^hello --> writes the blue hello completely opaque
^c#d0FF0000^hello --> writes red hello with 0xd0 transparency
(same for ^b^)

**Comment by bakkeby** [comment]
The implementation that is there has support for alpha, but it uses the alpha values for SchemeNorm.

Besides that it will affect everything else that uses SchemeNorm you would be able to play around with the alpha values there to see what the status2d text/drawing would look like. 

A very simple change would be to add additional markup options to change the colour scheme, or additional markup that allows you to set the alpha value for the next character. 

That said in the context of status2d markup it would make more sense to add support for RGBA values like you suggest (just that the alpha value would be at the end rather than at the beginning).

This may make more sense on top of the RGBA changes made on this experimental branch of dwm-flexipatch:
https://github.com/bakkeby/dwm-flexipatch/tree/vertigo

I shall have a look to see what I can do.

**Comment by bakkeby** [comment]
@Vinschers I added an example implementation of status2d with RGBA support on the aforementioned branch which you can have a play around with if you want. It should be feasible to backport this to the main dwm-flexipatch build.
